src/TileSlider/TileSlider.tsx

```diff
@@ -178,9 +186,12 @@ const TileSlider = <T extends unknown>({
         const movementX: number = Math.abs(newPosition.x - touchPosition.x);
         const movementY: number = Math.abs(newPosition.y - touchPosition.y);
 
-        if (movementX > movementY && movementX > 10) {
+        if ((movementX > movementY && movementX > 10) || verticalScrollBlockedRef.current) {
           event.preventDefault();
           event.stopPropagation();
+
+          if (verticalScrollBlockedRef.current) verticalScrollBlockedRef.current = true;
```

Comment:
This line does nothing (when `verticalScrollBlockedRef.current` is false, it isn't set to `true`).

```suggestion
          verticalScrollBlockedRef.current = true;
```